drivers/modem/modem_cellular.c

```diff
@@ -88,11 +89,17 @@ struct modem_cellular_data {
 	uint8_t cmux_transmit_buf[2 * CONFIG_MODEM_CELLULAR_CMUX_MAX_FRAME_SIZE];
 	struct modem_cmux_dlci dlci1;
 	struct modem_cmux_dlci dlci2;
+	struct modem_cmux_dlci dlci3;
+	struct modem_cmux_dlci dlci4;
 	struct modem_pipe *dlci1_pipe;
 	struct modem_pipe *dlci2_pipe;
+	struct modem_pipe *dlci3_pipe;
+	struct modem_pipe *dlci4_pipe;
 	uint8_t dlci1_receive_buf[CONFIG_MODEM_CELLULAR_CMUX_MAX_FRAME_SIZE];
 	/* DLCI 2 is only used for chat scripts. */
 	uint8_t dlci2_receive_buf[CONFIG_MODEM_CELLULAR_CHAT_BUFFER_SIZES];
+	uint8_t dlci3_receive_buf[CONFIG_MODEM_CELLULAR_CHAT_BUFFER_SIZES];
+	uint8_t dlci4_receive_buf[CONFIG_MODEM_CELLULAR_CHAT_BUFFER_SIZES];
```

Comment:
Actually, if the goal is only to expose DLCIs for whatever else to use, could we make it so that the DLCIs would first exist without receive buffer, and the actual user of every DLCI would attach a buffer to it? This would prevent coupling of the receive buffers to the code that manages the CMUX link.

Reply:
The issue is the layering of pipes, a user of a pipe, like a chat module, does not know if it is connected to a UART, or a DLCI channel, very much by design :) UART requires a transmit and receive buffer, DLCI only requires a receive buffer. It would be a layering violation for the user of a pipe to need to know how it is implemented, which is required if the user has to pass a buffer/buffers of appropriate sizes :)